Terminal.Gui/Views/Toplevel.cs

```diff
@@ -69,75 +69,21 @@ public Toplevel ()
     #region Subviews
 
     // TODO: Deprecate - Any view can host a menubar in v2
-    /// <summary>Gets or sets the menu for this Toplevel.</summary>
-    public MenuBar? MenuBar { get; set; }
+    /// <summary>Gets the latest <see cref="MenuBar"/> added into this Toplevel.</summary>
+    public MenuBar? MenuBar => (MenuBar?)Subviews?.LastOrDefault (s => s is MenuBar);
```

Comment:
For future cases like this, using the safe cast form (`thingThatMightBeNull as ExpectedType`) is a better pattern than an explicit cast.

Reply:
Also, you can use the generic form of the methods instead of explicit checking, and avoid the whole thing:

```csharp
    public MenuBar? MenuBar => Subviews?.OfType<MenuBar>().LastOrDefault ();
```